Dockerfile

```diff
@@ -1,6 +1,16 @@
+#export DOCKER_CLI_EXPERIMENTAL=enabled
+#docker run --rm --privileged docker/binfmt:66f9012c56a8316f9244ffd7622d7c21c1f6f28d
+#docker buildx create --use --name mybuilder
+#docker buildx build -t colek42/csi-node-driver-registrar --platform=linux/arm,linux/arm64,linux/amd64 . --push
```

Comment:
`Docker Buildx` is included in Docker 19.03 see https://docs.docker.com/buildx/working-with-buildx/ also we need to make sure that are we able to push this image to  quay.io as csi is using quay.io

Reply:
Looks like Quay supports manifest files

ref:  https://coreos.com/quay-enterprise/releases/#1.15.2
ref:  https://docs.docker.com/registry/spec/manifest-v2-2/

> * Does that also work in Prow or only in TravisCI? We wanted to move away from TravisCI, it would be a step back to add more reasons why we have to use TravisCI.

I don't understand Prow, where can I find more info about it?  If Prow has a current verion of Docker and allows us to use buildx it should work.

> * Where and how do we choose the Go version? Right now it's specified in exactly one place (release-tools/travis.yml) and more precise with major.minor.patch version than the `FROM golang:1.13` in the Dockerfile.

If we are building in the container, I think it makes sense to specify the Go version in the 
Dockerfile.  

> * Do we keep "make build" as-is (i.e. build with Go on the host) or do we replace it with building in Docker? This is mostly relevant for developers.

I always reccomend building in a container for reproducabilty.  If developers need the same functionality we can have the Makefile call `docker build` and mount a local folder into /bin

> * Do we need a `.dockerignore` file? Without it, `ADD . /app` may end up copying quite a bit of data from an unclear work directory when invoked by a developer. In PMEM-CSI, we blacklist everything and then whitelist just the parts that are needed for building (https://github.com/intel/pmem-csi/blob/devel/.dockerignore). This is sometimes a nuisance to maintain, though.

I think a blacklist makes sense.  We are using a multi-stage build which will only copy over the artifact created in the build container.

Reply:
> I don't understand Prow, where can I find more info about it? If Prow has a current verion of Docker and allows us to use buildx it should work.

buildx depends on QEMU for simple multi-platform support, right? That may create additional dependencies for the host that the job runs on. Prow is the CI for Kubernetes, running on Kubernetes itself: https://github.com/kubernetes/test-infra/blob/master/prow/README.md

> If we are building in the container, I think it makes sense to specify the Go version in the
Dockerfile.

I don't mind that. I was just pointing out that if we move it, then further work is needed to adapt the other code that currently reads the Go version from travis.yml. My concern also was that 1.13 is less precise than what we are currently using. I don't know whether it matters in practice, though.

Reply:
currently, quay.io doesn't support multi-arch see https://github.com/ceph/ceph-csi/pull/707#issuecomment-572088862.

Reply:
@pohly Since quay doesn't support multi-arch, could there be a possibility to move to gcr?

Reply:
Yes, that has been the goal for a while now. We haven't done it yet because it wasn't clear how Prow jobs would get access to the necessary credentials for pushing to gcr. I believe that has been sorted out now.

Can you check whether `docker buildx` really works in a Prow job for multiple architectures? You can change the `release-tools/build.make` for that (for example, let `test` depend on a new `test-buildx` which runs the command), then we'll see during the next `pull-kubernetes-csi-node-driver-registrar-unit` run whether it succeeds.

Reply:
i'd like to get multi-arch builds off of prow and onto the GCB based image auto push infra so as to not need to register quemu on the nodes (and avoid different jobs stomping it).

I'm personally too oversubscribed for this, but the CNCF infra is generally moving that way. @justaugustus and @katharine have some work here.